bin/utils/project-event-permissions.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -eu
+
+state_where_clause=""
+role_where_clause=""
+
+while getopts ":s:r:h" option; do
+  case ${option} in
+    s)
+      state_where_clause="AND s.reference='${OPTARG}'"
+      ;;
+    r)
+      role_where_clause="AND r.reference='${OPTARG}'"
+      ;;
+    h)
+      echo "
+        Usage: ${0} [-s state] [-r role]
+
+        Options:
+          -s: allows to narrow down the list to specific case state (optional)
+          -r: allows to narrow down the list to specific user role (optional)
+      "
+      exit 1
+      ;;
+    *)
+      ;;
+  esac
+done
+
+query="
+  SELECT COALESCE(s.reference, '*') as state,
+    r.reference as user_role,
+    e.reference as event_id,
+    e.name as event_name
+  FROM event e
+    LEFT JOIN event_pre_state eps ON e.id = eps.event_id
+    LEFT JOIN state s ON eps.state_id = s.id
+    JOIN event_acl ea ON e.id = ea.event_id
+    JOIN role r ON ea.role_id = r.id
+  WHERE e.case_type_id = (SELECT MAX(id) FROM case_type)
```

Comment:
I am not biggest fan of that MAX(id), as it's not going to work i we have multiple case types, but no biggie, maybe we could flag somewhere it's not supported for multiple case_types?

Reply:
Good point, I assumed we will have just one case type. I can add extra where clause so that only our case type is taken to consideration.

Generally when new case type is added that script should be changed anyway otherwise it will yield unexpected / confusing results.

Reply:
I also head idea to make it as an external tool that could visualise results in a better way. Such tool could contain other projectors that for example render case state progression. Let's say this is MVP. I am not sure how useful would extra tool be to be honest so it is just an idea in my head ;)

Reply:
Yeah good for now definitely! It improves our life :)